src/Uno.UI/UI/Xaml/Media/ImageSource.iOS.cs

```diff
@@ -190,44 +202,65 @@ internal async Task<UIImage> Open(CancellationToken ct)
 		/// </summary>
 		private async Task OpenUsingPlatformDownloader(CancellationToken ct)
 		{
-			if (SupportsAsyncFromBundle)
-			{
-				DownloadUsingPlatformDownloader();
-			}
-			else
+			if (ct.IsCancellationRequested)
 			{
-				await CoreDispatcher.Main.RunAsync(
-					CoreDispatcherPriority.Normal,
-					() =>
-					{
-						DownloadUsingPlatformDownloader();
-					}
-				).AsTask(ct);
+				return;
 			}
-		}
 
-		private void DownloadUsingPlatformDownloader()
-		{
 			using (var url = new NSUrl(WebUri.OriginalString))
 			{
-				NSError error;
-
-				if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+				using (var request = NSUrlRequest.FromUrl(url))
 				{
-					this.Log().Debug($"Loading image from [{WebUri.OriginalString}]");
-				}
+					NSUrlSessionDataTask task;
+					var awaitable = DefaultSession.CreateDataTaskAsync(request, out task);
+					ct.Register(OnCancel);
+					try
+					{
+						task.Resume(); // We need to call this manually https://bugzilla.xamarin.com/show_bug.cgi?id=28425#c3
+						var result = await awaitable;
+						task = null;
+						var response = result.Response as NSHttpUrlResponse;
+
+						if (ct.IsCancellationRequested)
+						{
+							return;
+						}
+						else if (!IsSuccessful(response.StatusCode))
+						{
+							if (this.Log().IsEnabled(LogLevel.Error))
```

Comment:
We're not doing this yet, but maybe we should add the raise of [ImageFailed](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.image.imagefailed).

Reply:
ImageFailed will be raised later if ImageData is null when Open() returns. It could be interesting to include the error in the event args, though.